ui/app/AppLayouts/Chat/components/ProfilePopup.qml

```diff
@@ -7,62 +7,195 @@ import "./"
+      anchors.leftMargin: Theme.smallPadding
       anchors.top: parent.top
+      anchors.topMargin: Theme.smallPadding
+    }
+
+    Text {
+      id: valueEnsName
+      text: "@emily.stateofus.eth"
+      font.pixelSize: 14
       anchors.left: parent.left
-      font.bold: true
-      font.pixelSize: 17
-      anchors.leftMargin: 16
-      anchors.topMargin: Theme.padding
-      anchors.bottomMargin: Theme.padding
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: labelEnsUsername.bottom
+      anchors.topMargin: Theme.smallPadding
     }
 
-    Rectangle {
-      id: profilePic
-      width: 120
-      height: 120
-      radius: 100
-      border.color: "#10000000"
-      border.width: 1
-      color: Theme.transparent
-      anchors.horizontalCenter: parent.horizontalCenter
-      anchors.top: parent.top
-      anchors.topMargin: 16
-      Image {
-          width: 120
-          height: 120
-          fillMode: Image.PreserveAspectFit
-          source: identicon
-      }
+    Text {
+      id: labelChatKey
+      text: qsTr("Chat key")
+      font.pixelSize: 13
+      font.weight: Font.Medium
+      color: Theme.darkGrey
+      anchors.left: parent.left
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: valueEnsName.bottom
+      anchors.topMargin: Theme.padding
     }
 
     Text {
-      id: userNameText
-      text: userName
-      anchors.top: profilePic.bottom
-      anchors.topMargin: 16
-      anchors.horizontalCenter: parent.horizontalCenter
-      font.bold: true
-      font.pixelSize: 16
+      id: valueChatKey
+      text: fromAuthor.substr(0, 14) + "..." + fromAuthor.substr(fromAuthor.length - 14)
+      font.pixelSize: 14
+      anchors.left: parent.left
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: labelChatKey.bottom
+      anchors.topMargin: Theme.smallPadding
+    }
+
+    Separator {
+      id: separator
+      anchors.top: valueChatKey.bottom
+      anchors.topMargin: Theme.padding
+      anchors.left: parent.left
+      anchors.leftMargin: -Theme.padding
+      anchors.right: parent.right
+      anchors.rightMargin: -Theme.padding
     }
 
-    TextEdit {
-      text: fromAuthor.substr(0, 6) + "..." + fromAuthor.substr(fromAuthor.length - 4)
-      anchors.top: userNameText.bottom
-      anchors.topMargin: 12
-      anchors.horizontalCenter: parent.horizontalCenter
-      wrapMode: Text.Wrap
-      readOnly: true
-      selectByMouse: true
+    Text {
+      id: labelShareURL
+      text: qsTr("Share Profile URL")
+      font.pixelSize: 13
+      font.weight: Font.Medium
       color: Theme.darkGrey
-      font.pixelSize: 15
+      anchors.left: parent.left
+      anchors.leftMargin: Theme.smallPadding
+      anchors.top: separator.bottom
+      anchors.topMargin: Theme.padding
+    }
+
+    Text {
+      id: valueShareURL
+      text: "https://join.status.im/u/" + fromAuthor.substr(0, 4) + "..." + fromAuthor.substr(fromAuthor.length - 5)
```

Comment:
Does the link work with the `...`?

Reply:
No, but frankly, it's not really a clickable link anyways (also by design AFAICS)

Reply:
Weird, I guess it was a mistake in the design, because a link that's no clickable and even un copiable seems very useless haha

Reply:
Well, I might be wrong, but the design doesn't indicate it's clickable. However, it should have a copy-to-clipboard button next to it, but that isn't there yet.